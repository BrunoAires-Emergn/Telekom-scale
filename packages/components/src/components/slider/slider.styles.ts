import { JssStyle } from 'jss';

const sliderKnob: JssStyle = {
  '-webkit-appearance': 'none',
  border: '2px solid #409eff',
  height: '20px',
  width: '20px',
  borderRadius: '50%',
  background: '#fff',
};

export const styles: JssStyle = {
  slider: {
    display: 'inline-flex',
    alignItems: 'center',
    width: '100%',
  },
  slider__text: {
    margin: 'var(--slider-text-margin, 0 0 0 1rem)',
    minWidth: 30,
    textAlign: 'right',
  },
  slider__input: {
    outline: 'none',
    '-webkit-appearance': 'none',
    cursor: 'pointer',
    background: '#e4e7ed',
    height: '4px',
    width: '100%',
    borderRadius: '2px',
    '&::-webkit-slider-thumb': sliderKnob,
    '&::-moz-range-thumb': sliderKnob,
    '&::-ms-thumb': sliderKnob,
  },
  "input[type='range']": {
    '&::-moz-focus-outer': {
      border: '0',
    },
  },
};
