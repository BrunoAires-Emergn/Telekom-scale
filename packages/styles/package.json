{
    "name": "@telements/styles",
    "version": "0.0.1",
    "main": "dist/index.js",
    "scripts": {
        "build": "tsc -b",
        "postbuild": "node ./dist/export-css.js",
        "setup": "yarn"
    },
    "author": "@eeegor <egor.kirpichev@telekom.de>",
    "license": "MIT",
    "devDependencies": {
        "typescript": "^3.4.5"
    },
    "dependencies": {

    }
}
