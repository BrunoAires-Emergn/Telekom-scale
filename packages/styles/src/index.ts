import { button } from './button';
import { alert } from './alert';
import { card } from './card';

const theme = {
	colors: {
		white: '#fff',
		black: '#000',
		blue: '#00f',
		red: '#f00',
	}
}

export {
	theme,
	alert,
	button,
	card,
}
