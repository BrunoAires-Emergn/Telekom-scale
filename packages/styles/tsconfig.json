{
	"extends": "../tsconfig.base",
	"compilerOptions": {
		"rootDir": "./src",
		"outDir": "./dist",
		"types": [
			"node",
			"jest"
		]
	}
}
