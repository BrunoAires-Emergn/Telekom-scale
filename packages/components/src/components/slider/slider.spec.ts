import { newSpecPage } from '@stencil/core/testing';
import { Slider } from './slider';
import { styles } from './slider.styles';
import jss from 'jss';

describe('Slider', () => {
  let element;
  let stylesheet;
  beforeEach(async () => {
    element = new Slider();
    stylesheet = element.stylesheet = jss.createStyleSheet(styles as any);
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Slider],
      html: `<t-slider>Label</t-slider>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  // it('should handle a custom css class', () => {
  //   element.customClass = 'custom';
  //   expect(element.getCssClassMap()).toContain('custom');
  // });

  // it('should have a default css class', () => {
  //   expect(element.getCssClassMap()).toContain('slider');
  // });
});
