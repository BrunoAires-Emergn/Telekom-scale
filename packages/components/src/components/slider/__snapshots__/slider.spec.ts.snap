// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should match snapshot 1`] = `
<t-slider>
  <mock:shadow-root>
    <style>
      .slider-0-0-5 {
  width: 100%;
  display: inline-flex;
  align-items: center;
}
.slider__text-0-0-6 {
  margin: var(--slider-text-margin, 0 0 0 1rem);
  min-width: 30px;
  text-align: right;
}
.slider__input-0-0-7 {
  width: 100%;
  cursor: pointer;
  height: 4px;
  outline: none;
  background: #e4e7ed;
  border-radius: 2px;
  -webkit-appearance: none;
}
.slider__input-0-0-7::-webkit-slider-thumb {
  width: 20px;
  border: 2px solid #409eff;
  height: 20px;
  background: #fff;
  border-radius: 50%;
  -webkit-appearance: none;
}
.slider__input-0-0-7::-moz-range-thumb {
  width: 20px;
  border: 2px solid #409eff;
  height: 20px;
  background: #fff;
  border-radius: 50%;
  -webkit-appearance: none;
}
.slider__input-0-0-7::-ms-thumb {
  width: 20px;
  border: 2px solid #409eff;
  height: 20px;
  background: #fff;
  border-radius: 50%;
  -webkit-appearance: none;
}
.input\\[type\\=\\'range\\'\\]-0-0-8::-moz-focus-outer {
  border: 0;
}
    </style>
    <div class="slider-0-0-5">
      <input class="slider__input-0-0-7" max="100" min="0" step="1" type="range" value="0">
    </div>
  </mock:shadow-root>
  Label
</t-slider>
`;
